core/trino-main/src/main/java/io/trino/metadata/Metadata.java

```diff
@@ -256,6 +256,13 @@ Optional<TableExecuteHandle> getTableHandleForExecute(
      */
     void dropTable(Session session, TableHandle tableHandle);
 
+    /**
+     * Drops the specified table
+     *
+     * @throws RuntimeException if the table cannot be dropped
+     */
+    void dropTable(Session session, QualifiedObjectName tableName);
```

Comment:
We want this method to replace the old one, `void dropTable(Session session, TableHandle tableHandle)`.
So it would be good to 

- make ConnectorMetadata change backward compatibility
- apply changes to `DropTableTask`

Doing this would reveal that the old drop handles redirections whereas the new drop-by-name cannot do that.
The MetadataManager could follow redirects as needed, but I am concerned this is not the right place to add that logic.

Also, decoupling `ConnectorMetadata.redirectTable` and `ConnectorMetadata.dropTable` calls makes it impossible to make a non-racy implementation.
I am leaving towards introducing a non-void return type here, so that drop-by-name can either declare success or return a redirection.

Reply:
@martint wdyt?

Reply:
This looks unresolved. @krvikash @findinpath do you have a plan here?

Reply:
perhaps  something like this (rough idea, not polished up)

```
DropResult dropTable(Session session, QualifiedObjectName tableName);
```

```java
sealed class DropResult
  permits DropSuccess, DropRedirected
```

```java
class DropSuccess {  .. }  // singleton (can this be an enum?)
```

```java
records DropRedirected(CatalogSchemaTableName target) {}
```